watchevents/services.go

```diff
@@ -0,0 +1,168 @@
+package watchevents
+
+import (
+	"bytes"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	kubernetesServiceKind = "kubernetesService"
+)
+
+type serviceHandler struct {
+	rClient          *client.RancherClient
+	kClient          *kubernetesclient.Client
+	serviceWatchChan chan struct{}
+}
+
+func NewServiceHandler(rClient *client.RancherClient, kClient *kubernetesclient.Client) *serviceHandler {
+	sHandler := &serviceHandler{
+		rClient: rClient,
+		kClient: kClient,
+	}
+	return sHandler
+}
+
+func (s *serviceHandler) startServiceWatch() chan struct{} {
+	watchlist := cache.NewListWatchFromClient(s.kClient.K8sClient.Core().RESTClient(), "services", v1.NamespaceAll, fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1.Service{},
+		time.Second*0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				key, _ := cache.MetaNamespaceKeyFunc(obj)
+				logrus.Infof("Received event: [ADDED] for Service: %s, Handling Add event.", key)
+				err := s.add(obj, "Added")
+				if err != nil {
+					logrus.Errorf("Error Handling event: [ADDED] for Service: %v", err)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				key, _ := cache.MetaNamespaceKeyFunc(obj)
+				logrus.Infof("Received event: [DELETED] for Service: %s, Handling Delete event.", key)
+				err := s.delete(obj)
+				if err != nil {
+					logrus.Errorf("Error Handling event: [DELETED] for Service: %v", err)
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				key, _ := cache.MetaNamespaceKeyFunc(newObj)
+				logrus.Infof("Received event: [MODIFIED] for Service: %s, Handling Modified event.", key)
+				err := s.add(newObj, "Modified")
```

Comment:
I think it would be better if we use constants for events types where those types are needed, instead of hardcoding them